AEG-19-JK: install and create basic page structure with react-router-dom

JamesAKing opened the pull request:
Ticket : https://aegis-dev.atlassian.net/jira/software/projects/AEG/boards/2?selectedIssue=AEG-19

- Removed unnecessary folders and files. 
- Added "react-router-dom" as dependency. 
- Set up routing
- Created router config files with all currently used urls. 
- Built temp header and nav bar for initial easy site navigation (Beware styles used)

JamesAKing committed:
removed unnecessary files and code

JamesAKing committed:
added pages folders and files

JamesAKing committed:
created routes and assigned route urls to variables

JamesAKing committed:
created basic header with functional page  links

JamesAKing committed:
created basic dashboard nav bar with funcitonal links

JamesAKing committed:
fixed bug where dashboard would disappear when navigating to routes

D-Nugent commented on src/components/dashboard-nav/DashboardNav.js:
```diff
@@ -0,0 +1,42 @@
+function DashboardNav() {
+    return (
+        <nav>
+            <ul>
```
Might need to see it visually but at a glance I don't believe that the `<ul> and <li>` elements are necessary here as nothing else is nested. So could perhaps reduce down to` <nav>'s and <NavLink's>`

JamesAKing replied:
Fair point. 
My reasoning for doing so is:
- Our cohort were taught this is a common convention.  
- It makes sense. Many sources refer to a "list of links". 
- All MDN examples for "pure links navs" have lists located inside their Nav Element [https://developer.mozilla.org/en-US/docs/Web/HTML/Element/nav](url).

JamesAKing replied:
Agreed to keep UL LI elements

D-Nugent commented on src/components/header/Header.js:
```diff
@@ -0,0 +1,34 @@
+    return (
+        <header>
+            <nav>
+                <ul style={{display: 'flex', listStyle: "none"}}>
```
Assume that later these styles will be added to their own stylesheet.

JamesAKing replied:
100% - temp measure to see what was going on.

D-Nugent commented on src/components/header/Header.js:
```diff
@@ -0,0 +1,34 @@
+                        <NavLink to={homeURL}>Home</NavLink>
+                    </li>
+                    <li style={{padding: "1rem"}}>
+                        <NavLink to={aboutURL}>About</NavLink>
```
Probably more to be added at a later date but might want to think about whether any links should not be visible depending on state. i.e. Dashboard probably shouldn't be visible unless a user is logged in and when Loged in there might be other tabs you don't want visible.

JamesAKing replied:
100% - temp measure to see what was going on / add easy nav around site.

D-Nugent commented on src/utilities/routerConfig.js:
```diff
@@ -0,0 +1,13 @@
+export const homeURL = '/';
```
Overall routes look good. At a later stage we may need to change the structure a bit to consider uid's. i.e. will we have '/profile/:profileId'. If routes get more convoluted as we progress we may want to consider passing routes into a singular object as necessary. i.e.:
```
export const routes = {
  dashboard:{
    url: '/dashboard';
    reports: '/dashboard/reports';
    foundItem: '/dashboard/found-item';
    profile: '/dashboard/profile';
  },
...etc
}

I think this will only really become necessary if we have a lot of sub-routes for pages.
```

JamesAKing replied:
Replies under your original comments below: 

When I switched to your branch I had some dependency conflicts:
- Added as dependency's.  

I didn't get the same error when attempting to install on main-branch (didn't do so for now). So this may require review later.
- Added as new bug with link to possible fix. 

Would suggest using '.jsx' or '.tsx' (if we end up using typescript) for our components as this gives some better intellisense when coding.
- I have no issue doing this but would love to hear your thoughts / expansion on intelluisense and why .jsx is your preferred choice. 

Small note: As a standard, I use PascalCase for folder names when they contain components. i.e. pages/Contact/Contact.jsx but honestly not necessary just a standard I've seen a lot.
- No issue with this as long as it is consistent throughout the site. 
- As I understand this is just a preference thing. I use kebab case as it clearly differentiates folder from file which (in my head) helps to easily navigate / find / differentiate in large projects.
- Proposal: in situations like this where it is preference and neither of us is wrong, it could be worth defaulting to a specific standard /rule book [(e.g. airbnb)
](https://github.com/airbnb/javascript/tree/master/react)

There's some boilerplate stuff we can still get rid of in the public folder still but can be done later when we update favicon etc.
- Sure thing. My work flow is exactly that. Touch this when I have the assets to replace or at time of production build.

JamesAKing replied:
Will swap to abort based method for URLs

D-Nugent reviewed (commented):
Other comments:
- When I switched to your branch I had some dependency conflicts:
    - sass: I assume you have it installed globally?
    - react-router-dom: didn't see this installed in the dependencies. When I attempted to install on your branch I received an error of: 
![image](https://user-images.githubusercontent.com/73457386/128469709-3adc52fe-b91f-4f00-aac4-ae1c7e829d47.png)
I didn't get the same error when attempting to install on main-branch (didn't do so for now). So this may require review later.
- Would suggest using '.jsx' or '.tsx' (if we end up using typescript) for our components as this gives some better intellisense when coding.
- Small note: As a standard, I use PascalCase for folder names when they contain components. i.e. pages/Contact/Contact.jsx but honestly not necessary just a standard I've seen a lot.
- There's some boilerplate stuff we can still get rid of in the public folder still but can be done later when we update favicon etc.
Overall, looks great! Thanks for doing this.

JamesAKing committed:
added sass and react router dom as dependencies

JamesAKing commented:
Replies under your original comments below:

When I switched to your branch I had some dependency conflicts:
- Added as dependency's.

I didn't get the same error when attempting to install on main-branch (didn't do so for now). So this may require review later.
- Added as new bug with link to possible fix.

Would suggest using '.jsx' or '.tsx' (if we end up using typescript) for our components as this gives some better intellisense when coding.
- I have no issue doing this but would love to hear your thoughts / expansion on intellisense and why .jsx is your preferred choice.

Small note: As a standard, I use PascalCase for folder names when they contain components. i.e. pages/Contact/Contact.jsx but honestly not necessary just a standard I've seen a lot.
- No issue with this as long as it is consistent throughout the site.
- As I understand this is just a preference thing. I use kebab case as it clearly differentiates folder from file which (in my head) helps to easily navigate / find / differentiate in large projects.
- Proposal: in situations like this where it is preference and neither of us is wrong, it could be worth defaulting to a specific standard /rule book (e.g. airbnb)

There's some boilerplate stuff we can still get rid of in the public folder still but can be done later when we update favicon etc.
- Sure thing. My work flow is exactly that. Touch this when I have the assets to replace or at time of production build.

JamesAKing commented:
Will go forward with PascalCase as this is how Airbnb do it.

JamesAKing committed:
updated react componet files to jsx

JamesAKing committed:
updated folder names inline with new project style guide

JamesAKing committed:
updated folder names inline with new project style guide - now saved
